src/features/chat-message/mappers/chat-message.mapper.ts

```diff
@@ -0,0 +1,64 @@
+import { ChatMessageResponseDto } from '@features/chat-message/dtos/response/chat-message.response-dto';
+import { Injectable } from '@nestjs/common';
+import { ChatMessageEntity } from '@features/chat-message/domain/chat-message.entity';
+import { ChatMessageProps } from '@features/chat-message/domain/chat-message.entity-interface';
+
+import { baseSchema } from '@src/libs/db/base.schema';
+import { CreateEntityProps } from '@src/libs/ddd/entity.base';
+import { Mapper } from '@src/libs/ddd/mapper.interface';
+import { z } from 'zod';
+
+export const chatMessageSchema = baseSchema.extend({
+  roomId: z.bigint(),
+  senderId: z.bigint(),
+  message: z.string().max(1000),
+  deletedAt: z.preprocess(
+    (val: any) => (val === null ? null : new Date(val)),
+    z.nullable(z.date()),
+  ),
+});
+
+export type ChatMessageModel = z.TypeOf<typeof chatMessageSchema>;
+
+@Injectable()
+export class ChatMessageMapper
+  implements
+    Mapper<ChatMessageEntity, ChatMessageModel, ChatMessageResponseDto>
+{
+  toEntity(record: ChatMessageModel): ChatMessageEntity {
+    const chatMessageProps: CreateEntityProps<ChatMessageProps> = {
+      id: record.id,
+      props: {
+        roomId: record.roomId,
+        senderId: record.senderId,
+        message: record.message,
+        deletedAt: record.deletedAt,
+      },
+      createdAt: record.createdAt,
+      updatedAt: record.updatedAt,
+    };
+
+    return new ChatMessageEntity(chatMessageProps);
+  }
+
+  toPersistence(entity: ChatMessageEntity): ChatMessageModel {
+    const { id, createdAt, updatedAt, ...props } = entity.getProps();
+
+    const record: ChatMessageModel = {
+      id,
+      createdAt,
+      updatedAt,
+      ...props,
+    };
+
+    return chatMessageSchema.parse(record);
```

Comment:
```suggestion
    const props = entity.getProps();

    const record: ChatMessageModel = props

    return chatMessageSchema.parse(record);
```
이렇게 하거나 그대로 props를 parse 메서드에 집어 넣어도 될 것 같은데 맞나요?